internal/librarian/release_please_lite.go

```diff
@@ -77,3 +81,31 @@ func formatTag(library *config.LibraryState) string {
 	r := strings.NewReplacer("{id}", library.ID, "{version}", library.Version)
 	return r.Replace(tagFormat)
 }
+
+// NextVersion calculates the next semantic version based on a slice of conventional commits.
+// If overrideNextVersion is not empty, it is returned as the next version.
+func NextVersion(commits []*gitrepo.ConventionalCommit, currentVersion, overrideNextVersion string) (string, error) {
+	if overrideNextVersion != "" {
+		return overrideNextVersion, nil
+	}
+	highestChange := getHighestChange(commits)
+	return semver.DeriveNext(highestChange, currentVersion)
+}
+
+// getHighestChange determines the highest-ranking change type from a slice of commits.
+func getHighestChange(commits []*gitrepo.ConventionalCommit) semver.ChangeLevel {
+	highestChange := semver.None
+	for _, commit := range commits {
+		if commit.IsBreaking {
+			highestChange = semver.Major
+			break // Major change has the highest precedence
+		}
+		if commit.Type == "feat" && highestChange != semver.Major {
+			highestChange = semver.Minor
```

Comment:
Since the change type is actually int, you can always set the type to the highest/lowest value and remove the `if` and `break`.